components/core/tests/test-end_to_end.cpp

```diff
@@ -0,0 +1,117 @@
+#include <sys/wait.h>
+
+#include <cstdlib>
+#include <filesystem>
+#include <string>
+#include <vector>
+
+#include <Catch2/single_include/catch2/catch.hpp>
+
+#include "../src/clp_s/JsonConstructor.hpp"
+#include "../src/clp_s/JsonParser.hpp"
+
+auto const cDefaultTargetEncodedSize = 8ULL * 1024 * 1024 * 1024;  // 8 GB
+auto const cDefaultMaxDocumentSize = 512ULL * 1024 * 1024;  // 512 MB
+auto const cDefaultMinTableSize = 1ULL * 1024 * 1024;  // 1 MB
+auto const cDefaultCompressionLevel = 3;
+auto const cDefaultPrintArchiveStats = false;
+auto const cDefaultOrdered = false;
+auto const cDefaultTargetOrderedChunkSize = 0;
+
+std::string const cTestEndToEndArchiveDirectory{"test-end-to-end-archive"};
+std::string const cTestEndToEndOutputDirectory = "test-end-to-end-out";
+std::string const cTestEndToEndOutputSortedJson = "test-end-to-end_sorted.json";
+std::string const cTestEndToEndInputFileDirectory = "test_log_files";
+std::string const cTestEndToEndInputFile = "test_no_floats_sorted.json";
```

Comment:
1. As mentioned in the [style guide](https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables), we shouldn't use static init for non-trivial objects. Any reason not to use a `constexpr string_view` as mentioned in our internal guidelines?
2. We should use brace-init for all of these.
3. Since these are only used in the one test, I would just put these directly in the test that they're used. When we add more tests later, we can easily move them out to be globals.

Reply:
3. These are used in the class and the 2 helper functions. Personally, I think it's cleaner to leave it at the top, so they don't need to be declared or passed to so many places.

Reply:
3. Sorry, this comment was supposed to apply to the variables above.